src/main/java/org/fsg1/fmms/backend/services/CurriculumService.java

```diff
@@ -39,8 +41,10 @@
      * @param parameters The first parameter should be the identifier of the curriculum.
      * @return A JSON ObjectNode of the resulting JSON object.
      */
-    public JsonNode execute(final Object... parameters) throws SQLException, IOException {
-        final ResultSet resultSet = getConn().executeQuery(getQueryString(), parameters[0]);
+    @Override
+    public JsonNode get(String query, Object... parameters) throws SQLException, IOException {
+        final ResultSet resultSet = getConn().executeQuery(getQueryCurriculumSemestersString(),
+                parameters[0]);
```

Comment:
There is no need to unpack the parameters var args. "executeQuery" takes a Object... as well.